test/jdk/jdk/incubator/vector/templates/Unit-Identity-test.template

```diff
@@ -0,0 +1,95 @@
+
+    @Test(dataProvider = "$type$UnaryOpProvider")
+    static void testIdentityValues(IntFunction<$type$[]> fa) {
+        $type$[] a = fa.apply(SPECIES.length());
+
+        for (int i = 0; i < a.length; i++) {
+            $type$ x = a[i];
+
+            // ADD identity: 0 + x == x
+            Assert.assertEquals(($type$)(ADD_IDENTITY + x), x,
+                                "ADD(ADD_IDENTITY, " + x + ") != " + x);
+#if[BITWISE]
+
+            // AND identity: -1 & x == x
+            Assert.assertEquals(($type$)(AND_IDENTITY & x), x,
+                                "AND(AND_IDENTITY, " + x + ") != " + x);
+#end[BITWISE]
+
+            // FIRST_NONZERO identity: firstNonZero(0, x) == x
+            Assert.assertEquals(firstNonZero(FIRST_NONZERO_IDENTITY, x), x,
+                                "FIRST_NONZERO(FIRST_NONZERO_IDENTITY, " + x + ") != " + x);
+
+            // MAX identity: max($Wideboxtype$.$MinValue$, x) == x
+            Assert.assertEquals(($type$) Math.max(MAX_IDENTITY, x), x,
+                                "MAX(MAX_IDENTITY, " + x + ") != " + x);
+
+            // MIN identity: min($Wideboxtype$.$MaxValue$, x) == x
+            Assert.assertEquals(($type$) Math.min(MIN_IDENTITY, x), x,
+                                "MIN(MIN_IDENTITY, " + x + ") != " + x);
+
+            // MUL identity: 1 * x == x
+            Assert.assertEquals(($type$)(MUL_IDENTITY * x), x,
+                                "MUL(MUL_IDENTITY, " + x + ") != " + x);
+#if[BITWISE]
+
+            // OR identity: 0 | x == x
+            Assert.assertEquals(($type$)(OR_IDENTITY | x), x,
+                                "OR(OR_IDENTITY, " + x + ") != " + x);
+
+            // SUADD identity: addSaturatingUnsigned(0, x) == x
+            Assert.assertEquals(($type$) VectorMath.addSaturatingUnsigned(SUADD_IDENTITY, x), x,
+                                "SUADD(SUADD_IDENTITY, " + x + ") != " + x);
+
+            // UMAX identity: maxUnsigned(0, x) == x
+            Assert.assertEquals(($type$) VectorMath.maxUnsigned(UMAX_IDENTITY, x), x,
+                                "UMAX(UMAX_IDENTITY, " + x + ") != " + x);
+
+            // UMIN identity: minUnsigned(-1, x) == x
+            Assert.assertEquals(($type$) VectorMath.minUnsigned(UMIN_IDENTITY, x), x,
+                                "UMIN(UMIN_IDENTITY, " + x + ") != " + x);
+
+            // XOR identity: 0 ^ x == x
+            Assert.assertEquals(($type$)(XOR_IDENTITY ^ x), x,
+                                "XOR(XOR_IDENTITY, " + x + ") != " + x);
+#end[BITWISE]
+        }
+    }
+
+    @Test(dataProvider = "$type$UnaryOpProvider")
+    static void testMaskedReductionIdentityAllFalse(IntFunction<$type$[]> fa) {
```

Comment:
Do we need this test since we already have boolean data sets with all false values that is used for the mask tests